keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -418,51 +395,43 @@ As a cluster administrator, I want to make GPUs from vendor ACME available to us
 of that cluster. I prepare the nodes and deploy the vendor's components with
 `kubectl create`.
 
-I create a ResourceClass for the hardware with parameters that only I as the
+I create a DeviceClass for the hardware with parameters that only I as the
 administrator am allowed to choose, like for example running a command with
-root privileges that does some cluster-specific initialization for each allocation:
+root privileges that does some cluster-specific initialization of a device
+each time it is prepared on a node:
 ```
-apiVersion: gpu.example.com/v1
-kind: GPUInit
-metadata:
-  name: acme-gpu-init
-# DANGER! This option must not be accepted for
-# user-supplied parameters. A real driver might
-# not even allow it for admins. This is just
-# an example to show the conceptual difference
-# between ResourceClass and ResourceClaim
-# parameters.
-initCommand:
-- /usr/local/bin/acme-gpu-init
-- --cluster
-- my-cluster
----
-apiVersion: core.k8s.io/v1alpha2
-kind: ResourceClass
+apiVersion: resource.k8s.io/v1alpha3
+kind: DeviceClass
 metadata:
   name: acme-gpu
-parametersRef:
-  apiGroup: gpu.example.com
-  kind: GPUInit
-  name: acme-gpu-init
+
+requirements:
+- deviceSelector: device.driverName == "gpu.acme.example.com"
+
+config:
+- opaque:
+    driverName: gpu.acme.example.com
+    parameters:
+      apiVersion: gpu.acme.example.com/v1
+      kind: GPUInit
+      # DANGER! This option must not be accepted for
+      # user-supplied parameters.
+      initCommand:
+        - /usr/local/bin/acme-gpu-init
+        - --cluster
+        - my-cluster
```

Comment:
So the kubelet plugin is now responsible for pulling down a device class, parsing it and deciding if this command is allowed or not?